internal/services/batch/batch_pool.go

```diff
@@ -59,6 +65,214 @@ func flattenBatchPoolFixedScaleSettings(settings *batch.FixedScaleSettings) []in
 	return append(results, result)
 }
 
+func flattenBatchPoolUserAccount(d *pluginsdk.ResourceData, account *batch.UserAccount) map[string]interface{} {
+	userAccount := make(map[string]interface{})
+	userAccount["name"] = *account.Name
+	userAccount["elevation_level"] = string(account.ElevationLevel)
+	userAccountIndex := -1
+	if num, ok := d.GetOk("user_accounts.#"); ok {
+		n := num.(int)
+		for i := 0; i < n; i++ {
+			if src, nameOk := d.GetOk(fmt.Sprintf("user_accounts.%d.name", i)); nameOk && src == *account.Name {
+				userAccount["password"] = d.Get(fmt.Sprintf("user_accounts.%d.password", i)).(string)
+				userAccountIndex = i
+				break
+			}
+		}
+	}
+	if account.LinuxUserConfiguration != nil {
+		linuxUserConfig := make(map[string]interface{})
+		if account.LinuxUserConfiguration.UID != nil {
+			linuxUserConfig["uid"] = *account.LinuxUserConfiguration.UID
+			linuxUserConfig["gid"] = *account.LinuxUserConfiguration.Gid
+		}
+		if userAccountIndex > -1 {
+			if sshPrivateKey, ok := d.GetOk(fmt.Sprintf("user_accounts.%d.linux_user_configuration.0.ssh_private_key", userAccountIndex)); ok {
+				linuxUserConfig["ssh_private_key"] = sshPrivateKey
+			}
+		}
+		userAccount["linux_user_configuration"] = []interface{}{
+			linuxUserConfig,
+		}
+	}
+	if account.WindowsUserConfiguration != nil {
+		loginMode := make(map[string]interface{})
+		loginMode["login_mode"] = string(account.WindowsUserConfiguration.LoginMode)
+		userAccount["windows_user_configuration"] = []interface{}{
+			loginMode,
+		}
+	}
+	return userAccount
+}
+
+func findSensitiveInfoForMountConfig(targetType string, sourceType string, sourceValue string, mountType string, d *pluginsdk.ResourceData) string {
+	if num, ok := d.GetOk("mount_configuration.#"); ok {
+		n := num.(int)
+		for i := 0; i < n; i++ {
+			if src, ok := d.GetOk(fmt.Sprintf("mount_configuration.%d.%v.0.%v", i, mountType, sourceType)); ok && src == sourceValue {
+				return d.Get(fmt.Sprintf("mount_configuration.%d.%v.0.%v", i, mountType, targetType)).(string)
+			}
+		}
+	}
+	return ""
+}
+
+func flattenBatchPoolMountConfig(d *pluginsdk.ResourceData, config *batch.MountConfiguration) map[string]interface{} {
```

Comment:
`d.Set` shouldn't be done in flatten methods - and all fields need to always be set into the state - can we update this?

Reply:
> `d.Set` shouldn't be done in flatten methods.
Ok, I need some time to do that conversion.
>  - and all fields need to always be set into the state - can we update this?
I am not sure what do you mean, we could always see `if xxx != nil {d.Set(xx, xx)}` in our code. Do you mean even a property is nil, we need to set it to be empty in `d.Set()`? Would it cause confusion between "user dont specify that property" and "user specify an empty property and it is ok"?